From: "Jiangyi Chen" <cjy520lcy@163.com>
Subject: [PATCH] win32u: Add match for second_name.
Message-Id: <2dae835a.4417.179456d7c7c.Coremail.cjy520lcy@163.com>
Date: Fri, 7 May 2021 14:04:33 +0800 (CST)

From c96631bc19deeb2465d9ffba4beaacfa0fe9e871 Mon Sep 17 00:00:00 2001
From: Jiangyi Chen <chenjiangyi@uniontech.com>
Date: Fri, 7 May 2021 14:01:13 +0800
#Subject: [PATCH] win32u: Add match for second_name. eg: there is a substituting
  MS Shell Dlg -> SimSun .the first parameter of GdipCreateFontFamilyFromName
 is MS Shell Dlg,then second_name is needed to find SimSun.Otherwise, it may
 cause GdipDrawString to crash indirectly.

Signed-off-by: Jiangyi Chen <chenjiangyi@uniontech.com>
---
 dlls/win32u/font.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/win32u/font.c b/dlls/win32u/font.c
index 11111111111..11111111111 100644
--- a/dlls/win32u/font.c
+++ b/dlls/win32u/font.c
@@ -3056,6 +3056,7 @@ static BOOL family_matches( struct gdi_font_family *family, const WCHAR *face_na
     struct gdi_font_face *face;
 
     if (!facename_compare( face_name, family->family_name, LF_FACESIZE - 1 )) return TRUE;
+    if (!facename_compare( face_name, family->second_name, LF_FACESIZE - 1 )) return TRUE;
     LIST_FOR_EACH_ENTRY( face, get_family_face_list(family), struct gdi_font_face, entry )
         if (!facename_compare( face_name, face->full_name, LF_FACESIZE - 1 )) return TRUE;
     return FALSE;
@@ -3165,7 +3166,7 @@ static BOOL CDECL font_EnumFonts( PHYSDEV dev, LOGFONTW *lf, FONTENUMPROCW proc,
             if (!family_matches(family, face_name)) continue;
             LIST_FOR_EACH_ENTRY( face, get_family_face_list(family), struct gdi_font_face, entry )
             {
-                if (!face_matches( family->family_name, face, face_name )) continue;
+                if (!face_matches( family->family_name, face, face_name ) && !face_matches( family->second_name, face, face_name )) continue;
                 if (!enum_face_charsets( family, face, enum_charsets, count, proc, lparam, orig_name ))
                     return FALSE;
 	    }

-- 
2.37.1

